src/todo.js

```diff
@@ -0,0 +1,170 @@
+import Task from './task';
+import allowNewline from './allow-newline';
+
+export default class {
+  constructor() {
+    this.tasks = localStorage.tasks ? JSON.parse(localStorage.tasks) : [];
+  }
+
+  addTask(description = '') {
+    const newEntry = document.getElementById('new-item');
+    if (description || newEntry.value) {
+      if (!description) description = newEntry.value;
+      newEntry.value = '';
+      this.tasks.push(new Task(description, this.tasks.length));
+      this.updateLocalStorage();
+      this.renderList();
+    }
+  }
+
+  updateIndexes() {
+    let i = 0;
+    this.tasks.forEach((task) => {
+      task.index = i;
+      i += 1;
+    });
+    this.updateLocalStorage();
+    this.renderList();
+  }
```

Comment:
Using two same lines of code many times, in many functions, makes your lines of code many, and also it makes your code DRY since they are duplicates which is what you do not want to happen 😞 . You can try to minimize these functions on line (15-16 & 26-27) into one global function, which can be reusable 🔂anywhere you wish to call it 😀.

Reply:
### Hello, @Ntwali-Josue!
Thanks for pointing that out this issue! :pray:  In this case, however, I would diverge from your point of view because, as I checked out, there's only one repetition of those lines, indeed...  Those lines aren't used anywhere else in my code...  On the other hand, sometimes I need one of the methods you noticed without the other... So, following the YAGNI principle, I think I'd better leave it as it is now...  Anyway, thanks for pointing it out: if I notice it comes to be repeated another time, I will definitely consider creating a function to encapsulate those methods! :+1:

Reply:
Sure @enionsouza thanks, for the clarifications 👍🏾